### PR TITLE
Click to deploy Kubeflow web app on GCP 

### PR DESCRIPTION
**Pull request by jlewi**
Create a react app to "click to deploy" on GCP.  …
* This is a minimal POC. The app provides a UI to help users
  create a deployment manager config to deploy Kubeflow.

* The app can then make request to GCP to actually create the deployment.

* The Google sign in button is used to facilitate logging into Google
  and getting Google credentials.

Related to #884 web app for deploying Kubeflow.

DO NOT MERGE until node_modules has been pruned. Its currently 245M and we don't want to merge anything that big.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/897)

### CODE REVIEW

File: components/gcp-click-to-deploy/src/App.test.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
Comment (yebrahim) [inline components/gcp-click-to-deploy/src/App.test.js]: Does this test work? Maybe we don't need one for now.

File: components/gcp-click-to-deploy/src/configs/env-kubeflow.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
```
Comment (yebrahim) [inline components/gcp-click-to-deploy/src/configs/env-kubeflow.sh]: Why do we need this file?

File: components/gcp-click-to-deploy/src/App.js [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+// TODO(jlewi): For the FQDN we should have a drop down box to select custom domain or automatically provisioned
+// domain. Based on the response if the user selects auto domain then we should automatically supply the suffix
+// <hostname>.endpoints.<Project>.cloud.goog
+
+var gapi = null;
+//const jinjaTemplate = "./configs/cluster.jinja";
+
+// TODO(jlewi): This is a hack to make variables available
+// in the debug console for further inspection.
+var globalResult = null;
```
Comment (yebrahim) [inline components/gcp-click-to-deploy/src/App.js]: This isn't being used anywhere.

File: components/gcp-click-to-deploy/package.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "click-to-deploy",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "gapi-client": "0.0.3",
```
[the remark this answers is not in the document]
Comment (jlewi) [reply components/gcp-click-to-deploy/package.json]: Done.

File: components/gcp-click-to-deploy/src/App.test.js (hunk 1 above)
Comment (jlewi) [reply components/gcp-click-to-deploy/src/App.test.js]: I fixed the test.

File: components/gcp-click-to-deploy/src/configs/env-kubeflow.sh (hunk 2 above)
Comment (jlewi) [reply components/gcp-click-to-deploy/src/configs/env-kubeflow.sh]: We don't; side effect of copying docs/gke/configs over.
I removed it.

File: components/gcp-click-to-deploy/src/App.js (hunk 3 above)
Comment (jlewi) [reply components/gcp-click-to-deploy/src/App.js]: Removed.

**Comment by jlewi** [comment]
Thanks. PTAL.

**Comment by jlewi** [comment]
@yebrahim Can you do "/lgtm" Prow requires that.
/approve

**Comment by yebrahim** [comment]
I can prowly get used to that.
/lgtm

**Comment by yebrahim** [comment]
.. but I guess that's not enough?

**Comment by jlewi** [comment]
@yebrahim I need to invite you to the GitHub org. I sent you an invite can you accept and re-lgtm.

**Comment by yebrahim** [comment]
Just did, thanks. 
/lgtm

**Pull request merged**